simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/evse/states/StoppedState.java

```diff
@@ -49,9 +53,11 @@ public void onUnplug(int evseId, int connectorId) {
 
                 stationMessageSender.sendTransactionEventEnded(evseId, connectorId, TransactionEventRequest.TriggerReason.EV_DEPARTED, TransactionData.StoppedReason.EV_DISCONNECTED);
             }
+            future.complete(UserMessageResult.SUCCESSFUL);
         });
 
         stateManager.setStateForEvse(evseId, new AvailableState());
+        return future;
```

Comment:
if the future os not successful, what would it be then? is it OK?

Reply:
If it is not successful then it is because everon did not reply to the message. If i send authorize for example and i don't get a response then the future will be not completed.

Reply:
you'd better be consistent and and define return values properly 
today everon does not use it, if/when we would -  no surprises... 

this is kinda minor, up to you

Reply:
Yes but how do i know if everon did not reply to the message, how can i complete the future to so error value if i can't know for sure when or if everon will reply?

Reply:
And the idea was that integration tests will be using these futures :)

Reply:
Does stationMessageSender.sendStatusNotificationAndSubscribe block and wait? 
What happens if connection is broken and some exception is thrown?

Reply:
No it does not block, the subscriber is saved in SubscriptionRegistry (using call id to match with response) class and then when we get the response it is called.
So if there is an exception or connection broken, the future will never get completed